.ado/templates/msbuild-sln.yml

```diff
@@ -50,7 +50,6 @@ steps:
         /p:PlatformToolset=${{parameters.platformToolset}}
         /p:PublishToolDuringBuild=true
         /p:EnableSourceLink=true
-        /p:RestoreLockedMode=true
```

Comment:
Why is this removed?

Reply:
Doesn't play well with PackageReference.

Is this prop critical?
Is it used for the NuGet caching?

Reply:
I ended up removing caching, but that prop is actually specific to PackageReference. It enforces that any change which alters the dependency chain also checks in the updated lockfile. See here for some details https://devblogs.microsoft.com/nuget/enable-repeatable-package-restores-using-a-lock-file/

If you see errors with it, it usually just means you need to rebuild the solution and check in the resulting packages.lock.json file.

Reply:
Adding a review task to the pre-merge checklist.

Reply:
In the current state, that argument makes restoring the managed C# project fail.
This could be related to inconsistent dependencies.
Looking into it.

```
msbuild.exe .\Microsoft.ReactNative.Managed\ /restore /p:RestoreLockedMode=true /t:Restore

...
Build FAILED.

"R:\pkr\vnext\Microsoft.ReactNative.Managed\Microsoft.ReactNative.Managed.csproj" (Restore target) (1) ->
(Restore target) ->
  R:\pkr\vnext\Microsoft.ReactNative.Managed\Microsoft.ReactNative.Managed.csproj : error NU1004: The project Microsoft.ReactNative has no compatible target framework. The packages lock file is inconsistent with the project dependencies so restore can't be run in locked mode. Di
sable the RestoreLockedMode MSBuild property or pass an explicit --force-evaluate option to run restore to update the lock file.

    0 Warning(s)
    1 Error(s)
```

Reply:
Update. Setting `RestoreLockedMode` to `false` defaults to overwriting the lock file with its own computed closure.
This is the diff for the managed project's updated lock file:
![image](https://user-images.githubusercontent.com/4507319/142953014-f3966d2d-677d-42b6-bdeb-21989269afeb.png)

Something that jumps immediately is `Common`, `fmt`, and `folly` as dependencies with version `1.0.0`.
Those are actually native projects that should be project dependencies of MSRN's main vcxproj, and not package dependencies for the Managed C# projection (I see no reference to those projects in the C# project file).
However, this may not matter because the failing case stays the same even with the updated lock file.

Reply:
Setting `/p:RestoreForceEvaluate=true` resolves this.

https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files